t/socket-tcp-setoption.t

```diff
@@ -0,0 +1,725 @@
+tcp-nodelay change from 0 to 1
+tcp-nodelay change from 1 to 0
+--- no_error_log
+[error]
+
+
+
+=== TEST 12: tcp-nodelay set by ~0/0.
+--- config
+    set $port $TEST_NGINX_SERVER_PORT;
+
+    location /t {
+        content_by_lua_block {
+            require "resty.core.socket"
+
+            local port = ngx.var.port
+            local sock = ngx.socket.tcp()
+            local ok, err = sock:connect("127.0.0.1", port)
+            if not ok then
+                ngx.say("failed to connect: ", err)
+                return
+            end
+            local v1, err = sock:getoption("tcp-nodelay")
+            if not v1 then
+                ngx.say("get default tcp-nodelay failed: ", err)
+                return
+            end
+
+            ok, err = sock:setoption("tcp-nodelay", 10)
+            if not ok then
+                ngx.say("enable tcp-nodelay failed: ", err)
+                return
+            end
+            local v2, err = sock:getoption("tcp-nodelay")
+            if not v2 then
+                ngx.say("get enabled tcp-nodelay failed: ", err)
+                return
+            end
+            ngx.say("tcp-nodelay change from ", v1, " to ", v2)
+
+            ok, err = sock:setoption("tcp-nodelay", 0)
+            if not ok then
+                ngx.say("disable tcp-nodelay failed: ", err)
+                return
+            end
+            local v3, err = sock:getoption("tcp-nodelay")
+            if not v3 then
+                ngx.say("get disabled tcp-nodelay failed: ", err)
+                return
+            end
+            ngx.say("tcp-nodelay change from ", v2, " to ", v3)
+
+            sock:close()
+        }
+    }
+--- request
+GET /t
+--- response_body
+tcp-nodelay change from 0 to 1
+tcp-nodelay change from 1 to 0
+--- no_error_log
+[error]
+
+
+
+=== TEST 13: sndbuf.
+--- config
+    set $port $TEST_NGINX_SERVER_PORT;
+
+    location /t {
+        content_by_lua_block {
+            require "resty.core.socket"
+
+            local port = ngx.var.port
+            local sock = ngx.socket.tcp()
+            local ok, err = sock:connect("127.0.0.1", port)
+            if not ok then
+                ngx.say("failed to connect: ", err)
+                return
+            end
+            local v1, err = sock:getoption("sndbuf")
+            if not v1 then
+                ngx.say("get default sndbuf failed: ", err)
+                return
+            end
+
+            ok, err = sock:setoption("sndbuf", 4096)
+            if not ok then
+                ngx.say("enable sndbuf failed: ", err)
+                return
+            end
+            local v2, err = sock:getoption("sndbuf")
+            if not v2 then
+                ngx.say("get enabled sndbuf failed: ", err)
+                return
+            end
+            ngx.say("sndbuf change from ", v1, " to ", v2)
+
+            sock:close()
+        }
+    }
+--- request
+GET /t
+--- response_body_like eval
+qr/sndbuf change from \d+ to \d+/
```

Comment:
This change is too permissive. You should use anchors like `\A` and `\z` in the regex to make sure there's no unexpected output before this line of output.

Fix other similar places.

Reply:
accept